dali/operators/image/crop/random_crop_attr.h

```diff
@@ -28,6 +28,24 @@
 
 namespace dali {
 
+class RandomCropGeneratorWrap {
```

Comment:
Yes, essentially, I wanted to keep it similar to the lambda that was there, but to with the option to get the RNG. 

The underlying generator can be accessed as well if we kept a `std::vector` of them in the RandomCropAttr, which might be a better solution. Do you like it better?

Refactoring the interface seems outside of the scope of this task, but I am open to suggestions.

Reply:
I don't know if that's really out of scope. It's not a public interface and we clearly have a mismatch between the two interfaces (the function type and the RandomCropGenerator).

Reply:
I agree with Michał (outstanding pun) that it's a bit out of scope. Can you add add a jira task that would describe the expected refactor, @mzient? Then, we can make it a follow-up - as you said it's not public API so doing this now later makes little difference to users, while it can delay work on the checkpointing in general.